mktrayicon.c

```diff
@@ -216,26 +212,34 @@ int main(int argc, char **argv)
 	char *start_icon = "none";
 	FILE *fifo;
 	GThread *reader;
-
 	XInitThreads(); /* see http://stackoverflow.com/a/18690540/472927 */
 	gtk_init(&argc, &argv);
 
-	if (argc == 4 && strcmp(argv[1], "-i") == 0) {
-		start_icon = argv[2];
-	}
-
 	if (argc == 1) {
-		printf("Usage: %s [-i ICON] FIFO\n", *argv);
-		printf("Listen to FIFO for system tray icon specifications\n");
+		printf("Usage: %s [-i ICON] FIFO --or-- %s -i ICON -t TOOLTIP\n", *argv, *argv);
+		printf("Create a system tray icon as specified\n");
 		printf("\n");
 		printf("  -i ICON\tUse the specified ICON when initializing\n");
+		printf("  -t TOOLTIP\tUse the specified TOOLTIP when initializing\n");
 		printf("\n");
 		printf("Report bugs at https://github.com/jonhoo/mktrayicon\n");
 		return 0;
 	}
 
+	if (strcmp(argv[1], "-i") == 0) {
+		start_icon = argv[2];
+	}
+
 	icon = create_tray_icon(start_icon);
-	reader = g_thread_new("watch_fifo", watch_fifo, argv[argc-1]);
+
+	if (strcmp(argv[3], "-t") == 0) { /* icon is non-interactive, use given tooltip */
+		gtk_status_icon_set_tooltip_text(icon, argv[4]);
```

Comment:
We should error if `argv[4]` isn't set.

Reply:
If argv[3] is -t not followed by a string, it still works. Less typing to leave blank than to type `""`